### ISSUE TITLE
how to create master detail table? 

### ISSUE THREAD

**Issue by Karacraft** [ACFBentveld/Laravel-datatables#11]
I have followed datatables and yajra data tables (ditched it , due to very slow server side processing) tutorials and was successful in creating Master/Detail table. When i tried to apply the same in Laravel-datatables, it isn't working. 

First Problem is , getting that first empty column.
Here is the link [https://datatables.net/examples/api/row_details.html](https://datatables.net/examples/api/row_details.html)
In Html you need an empty 
`<td></td>`
in javascript you add following in columns:[]
```js
 {
        "className":      'details-control',
        "orderable":      false,
        "data":           null,
        "defaultContent": ''
 },
```

But nothing is working. 
Following error is being raised [https://datatables.net/manual/tech-notes/7](https://datatables.net/manual/tech-notes/7)

My Controller Method
```php
public function index()
{
    \DataTables::model(new CostCenter)->get();
    return view('avt.cost_centers.index');
}
```
My Javascript
```js
$(function() {
    table = $('#costcenterTable').DataTable({
        processing:true,
        serverSide:true,
        // stateSave: true,
        // fixedWidth:false,
        pagingType: "full_numbers",
        // scrollY:"100vh",
        scrollCollapse: false,
        //Languagte to add comma,deciaml
        language: {
        //    "decimal": ".",
            // "thousands": ",",
            lengthMenu: "Display _MENU_ records per page",
        },
        // ajax:location.href + "/data/getactivedata",
        ajax:location.href,
        columns:[
            {
                "className":'details-control',
                "orderable":false,
                "searchable":false,
                "data":null,
                "defaultContent": ''
            },
            { data: 'id' , name: 'id'},
            { data: 'acctcode' , name: 'acctcode'},
            { data: 'title' , name: 'title'},
            // { data: 'amount' , name: 'amount'},
            { data: 'amount' , name: 'amount',
                render: $.fn.dataTable.render.number( ',', '.', 0, '' )
            },
            { data: 'startyear' , name: 'startyear',
                render: function(data,type,full){ return moment(data).format('YYYY');}
            },
            { data: 'endyear' , name: 'endyear',
                render: function(data,type,full){ return moment(data).format('YYYY');}
            },
            { data: 'active' , name: 'active', sortable:false,orderable : false,
                render: function(data,type,row)
                {
                    // console.log(data);
                    // console.log(row);
                    return data == 1 ? '<i class="fa fa-check text-success"></i>' : '<i class="fa fa-times text-danger"></i>';
                }
            },
            { data: 'updated_at' , name: 'updated_at',
                render: function(data,type,full){ return moment(data).format('DD-MM-YYYY');}
            },
            // { data: 'action' , name: 'action', orderable:false,searchable:false},
        ],
        
    });
```

**Thanks in advance**

**Comment by wimurk** [comment]
@Karacraft i formatted your code. And i suggest that when you create an issue, please provide a readable piece of code.

Below is your javascript
```javascript
$(function() {
    table = $('#datatable').DataTable({
        processing:true,
        serverSide:true,
        pagingType: "full_numbers",
        scrollCollapse: false,
        ajax:location.href,
        language: {
            lengthMenu: "Display MENU records per page",
        },
        columns:[{
            "className":'details-control',
            "orderable":false,
            "searchable":false,
            "data":null,
            "defaultContent": ''
        },{
            data: 'id',
            name: 'id'
        },{
            data: 'acctcode',
            name: 'acctcode'
        },{
            data: 'title',
            name: 'title'
        },{
            data: 'amount',
            name: 'amount',
            render: $.fn.dataTable.render.number( ',', '.', 0, '' )
        },{
            data: 'startyear',
            name: 'startyear',
            render: function(data,type,full){
                return moment(data).format('YYYY');
            }
        },{
            data: 'endyear',
            name: 'endyear',
            render: function (data, type, full) {
                return moment(data).format('YYYY');
            }
        },{
            data: 'active',
            name: 'active',
            sortable: false,
            orderable: false,
            render: function (data, type, row) {
                return data == 1 ? '' : '';
            }
        },{
            data: 'updated_at',
            name: 'updated_at',
            render: function (data, type, full) {
                return moment(data).format('DD-MM-YYYY');
            }
        }]
    });
});
```
Your table will looks something like this
```html
<table id="datatable" class="table">
    <thead>
           <tr>
               <th>#</th>
               <th>id</th>
               <th>acctcode</th>
               <th>title</th>
               <th>amount</th>
                <th>startyear</th>
                <th>endyear</th>
                <th>active</th>
                <th>updated_at</th>
            </tr>
      </thead>
</table>
```

The code above triggers an error something like this
```sql
"SQLSTATE[42S22]: Column not found: 1054 Unknown column '' in 'order clause' (SQL: select * from `table` order by `` asc)"
```
This is because the first column you define has the name null and data is null. This can be fixed by removing the first column.

**Comment by wimurk** [comment]
In version 2.0.21 this issue will be fixed see merge request #12

**Comment by Karacraft** [comment]
Thanks for the reply.I'll ensure, next time i provide the code properly.

The column is in accordance to tutorials at datatables.net to show a child table or Data. 
So, can we not show a null column?

Also, I've upgraded to 2.0.21 and it is throwing the following error;

```php 
Argument 1 passed to ACFBentveld\DataTables\DataTables::filterColumns() must be of the type array, null given, called in C:\Dev\Server\data\htdocs\gen-inventory\vendor\acfbentveld\laravel-datatables\src\DataTables.php on line 142
```

**Comment by wimurk** [comment]
@Karacraft this is fixed in #13 version [v2.0.22](https://github.com/ACFBentveld/Laravel-datatables/releases/tag/v2.0.22)
